sqlglot/transforms.py

```diff
@@ -27,9 +29,12 @@ def preprocess(
     def _to_sql(self, expression: exp.Expression) -> str:
         expression_type = type(expression)
 
-        expression = transforms[0](expression)
-        for transform in transforms[1:]:
-            expression = transform(expression)
+        try:
+            expression = transforms[0](expression)
+            for transform in transforms[1:]:
+                expression = transform(expression)
+        except UnsupportedError as unsupported_error:
+            self.unsupported(str(unsupported_error))
```

Comment:
Did you test that this actually logs the correct warning? I don't remember if `str(...)` is the correct way to get the message.

Reply:
Yep, here's an example:

```
>>> sqlglot.parse_one('SELECT * FROM example_table CROSS JOIN UNNEST(split(type, ";"), scores)', read="starrocks").sql("hive")
Cannot transpile UNNEST with multiple arrays
"SELECT * FROM example_table CROSS JOIN EXPLODE(SPLIT(type, CONCAT('\\\\Q', ';')), scores, unnest(unnest))"
```

Reply:
Cool thanks for double checking